### PR TITLE
drivers: entropy: gecko: add entropy driver based on GECKO TRNG

### PR DESCRIPTION
**Pull request by pkaranjekar**
Add entropy driver based on GECKO TRNG module along with device
tree support for EFM32PG and EFR32MG SOCs.

Tested on boards EFM32PG-STK3402A and EFR32MG-SLTB004A with
samples/drivers/entropy application and tests/drivers/entropy test.

Signed-off-by: Pooja Karanjekar <pooja.karanjekar@lemonbeat.com>

### CODE REVIEW

**Comment by chrta** [comment]
The build error seems to indicate an issue with a cortex_r5 test and is not related to this pr:

```
INFO    - 937/937 qemu_cortex_r5            tests/kernel/context/kernel.common                 FAILED Failed (qemu 6.325s)
```

**Review by chrta — APPROVED** [review]
Also tested successfully on the EFM32PG-STK3402A board.

**Review by ceolin — APPROVED** [review]
Can you implement `get_entropy_isr` as well ?

**Review by dleach02 — APPROVED** [review]

File: dts/bindings/rng/gecko,trng.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2020, Lemonbeat GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    GECKO TRNG (True Random Number Generator).
+
+    The TRNG module is a non-deterministic randomnumber generator based on a full hardware solution.
+
+compatible: "silabs,gecko-trng"
```
Comment (MaureenHelm) [inline dts/bindings/rng/gecko,trng.yaml]: Please rename this file to match the compatible

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**